### ISSUE TITLE
Rebalance iteration numbers and include others

### ISSUE THREAD

**Issue by jburse** [jburse/dogelog-moon#147]
With the recent inclusion of eval_term(), Dogelog runtime now beats
Scryer Prolog in performance. But our benchmark iteration numbers do
not reflect the new speed thanks to eval_term(). They were developed when

we had an is/2 implemented via (=..)/2 and call/1. We shoudl rebalance the
benchmark, make each of the tests contribute equaly to the overal 
result. And then compare again with Scryer Prolog. We might now also

attempt a comparison with SWI and Jekejeke, although most likely we
are not yet there. Possibly Dogelog runtime will be still much slower 
then SWI and Jekejeke. Comparing with ECLiPSe Prolog or 

SWI-Prolog is possibly totally out of reach rigjt now.

**Comment by jburse** [comment]
After rebalance Dogelog runtime is 36% faster:

![bench_rebal_small](https://user-images.githubusercontent.com/5411487/132129893-7c768de2-7962-40d8-88e2-dfad2682e4c8.png)

The raw data is here:
[https://gist.github.com/jburse/f4e774ebb15cac722238b26b1a620f84#gistcomment-3882930](https://gist.github.com/jburse/f4e774ebb15cac722238b26b1a620f84#gistcomment-3882930)

**Comment by jburse** [comment]
We could also make a comparison between SWI-Prolog and Jekejeke Prolog.

We can work with this data to draw conclusions about what we did good
in Jekejeke Prolog and what we did good in Dogelog runtime, and about
the fundamental improvements that are still demanded for both Prolog systems.

Currently Dogelog runtime is ca. 4 times slower than Jekejeke Prolog.

![swi_jeke_small](https://user-images.githubusercontent.com/5411487/132130394-e490dd87-408d-4f38-b8cd-4a8304e32555.png)
